src/naming/main.cairo

```diff
@@ -699,9 +708,9 @@ mod Naming {
 
         // ADMIN
 
-        fn set_admin(ref self: ContractState, new_admin: ContractAddress) {
+        fn update_admin(ref self: ContractState, new_admin: ContractAddress) {
             assert(get_caller_address() == self._admin_address.read(), 'you are not admin');
-            self._admin_address.write(new_admin);
+            self.ownable.initializer(new_admin);
```

Comment:
_admin_address can be written to zero after this call so we can remove the variable in part 2